### PR TITLE
fix broken map-class links in docs

### PR DESCRIPTION
**Pull request by sbma44**
this guy:

image

### CODE REVIEW

**Comment by jfirebaugh** [comment]
Thanks @sbma44. See #661 for backstory here. Two things:
- Link should be to `l-map-class` (`L.Map`) rather than l-mapbox-map (`L.mapbox.Map`).
- 0200-01-01-l-tilelayer.html is a generated file and will be overwritten if we regenerate the docs. The fix needs to be in the generation code, [here](https://github.com/mapbox/mapbox.js/blob/14d4632fc8613d07a039bf82ffd162a05492b05c/_docs/generate.js#L95) probably.

**Comment by sbma44** [comment]
Thanks for the eyes, @jfirebaugh -- I was worried about not understanding the build process here, and seemingly with good reason. I'll get this building locally & figure out the appropriate fix.

**Comment by sbma44** [comment]
chatted with @lyzidiamond about this -- she's going to try to catch this class of errors in a more comprehensive manner & we'll proceed with a fix from there

**Comment by lyzidiamond** [comment]
@jfirebaugh @sbma44 Okay, finally fixed these!

boo regex yay regex <-- how i feel right now

**Comment by sbma44** [comment]
Thanks so much for grabbing this. @jfirebaugh lmk if I should figure out how to build this for a final :eyes: ; guessing you might have the context to hit merge more quickly than I would.

File: docs/_posts/api/0200-01-01-v2.2.1-all.html [hunk 1]
```diff
@@ -151,7 +151,7 @@ <h2 id="l-mapbox-gridlayer">L.mapbox.gridLayer(id|url|tilejson, options)</h2>
 </table>
 <p><em>Example</em>:</p>
 <pre><code>// the second argument is optional
-var layer = L.mapbox.gridLayer(&#39;mapbox.streets&#39;);
+var layer = L.mapbox.gridLayer(&#39;examples.map-20v6611k&#39;);
```
Comment (jfirebaugh) [inline docs/_posts/api/0200-01-01-v2.2.1-all.html]: You didn't intend to revert back to an examples account map here, right?

Comment (lyzidiamond) [inline docs/_posts/api/0200-01-01-v2.2.1-all.html]: well this whole example is wrong. `examples.map-20v6611k` is `mapbox.light` but neither of them are a `gridLayer()`... I will fix.

Comment (lyzidiamond) [inline docs/_posts/api/0200-01-01-v2.2.1-all.html]: Reading through, these docs need some additional love w/r/t v3 and `examples`. I'm going to fix these instances and merge, but then go through and do some additional cleanup in a different PR.

**Comment by lyzidiamond** [comment]
Per chat with @sbma44, I am going to merge here. We can roll back if we see anything glaring.

**Comment by sbma44** [comment]
thank you so much for tackling this, @lyzidiamond
